python/ray/util/iter.py

```diff
@@ -198,6 +198,45 @@ def for_each(self, fn: Callable[[T], U]) -> "ParallelIterator[U]":
         return self._with_transform(lambda local_it: local_it.for_each(fn),
                                     ".for_each()")
 
+    def for_each_concur(self,
```

Comment:
Not a big fan of this name but don't have a much better suggestion ATM. Would `for_each_pipelined` make sense?

Reply:
Consider `for_each_concurrent`, or simply extending the argument signature of for_each().

Reply:
Hmmm I'm open to a better name, but I think this function is doing concurrency, not pipeline parallelism.